test/system/StringTest.ooc

```diff
@@ -0,0 +1,155 @@
+/* This file is part of magic-sdk, an sdk for the open source programming language magic.
+ *
+ * Copyright (C) 2016 magic-lang
+ *
+ * This software may be modified and distributed under the terms
+ * of the MIT license.  See the LICENSE file for details.
+ */
+
+use unit
+
+StringTest: class extends Fixture {
+	init: func {
+		super("String")
+		this add("length, substring", func {
+			string := "12345678"
+			expect(string length(), is equal to(8))
+			substring := string substring(4)
+			substring2 := string substring(1, 3)
+			expect(substring, is equal to("5678"))
+			expect(substring2, is equal to("23"))
+			(substring, substring2) free()
+		})
+		this add("clone", func {
+			string := "123456"
+			clone := string clone()
+			other := clone clone()
+			expect(string, is equal to(clone))
+			clone free()
+			expect(other, is equal to(string))
+			other free()
+		})
+		this add("times, append, prepend", func {
+			string := "123"
+			times := string times(3)
+			expect(times, is equal to("123123123"))
+			times free()
+
+			prepend := string prepend("0")
+			append := prepend append("45")
+			expect(append, is equal to("012345"))
+			(prepend, append) free()
+		})
```

Comment:
@davidpiuva If you mean `prepend` and `append` leak, then yes, I already mentioned that in a comment above.